flake.nix

```diff
@@ -0,0 +1,43 @@
+{
+  description = "Lychee - A fast, async, stream-based link checker";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+    fenix.url = "github:nix-community/fenix";
+  };
+
+  outputs = {
+    self,
+    nixpkgs,
+    flake-utils,
+    fenix,
+    ...
+  }:
+    flake-utils.lib.eachDefaultSystem (system: let
+      pkgs = import nixpkgs {inherit system;};
+      toolchain = fenix.packages.aarch64-darwin.fromToolchainFile {
```

Comment:
How can we make it compatible with other platforms, such as `x86_64-linux` (my platform)

Reply:
The flake already does it, each default system includes Linux and Mac , arm and x86

Reply:
No, you currently hardcode the system here. It should be `fenix.packages.${system}.fromToolchainFile`.